Include/internal/pycore_interp.h

```diff
@@ -166,7 +166,17 @@ struct _is {
     struct _warnings_runtime_state warnings;
     struct atexit_state atexit;
 
-    struct _obmalloc_state obmalloc;
+    // Per-interpreter state for the obmalloc allocator.  For the main
+    // interpreter and for all interpreters that don't have their
+    // own obmalloc state, this points to the static structure in
+    // obmalloc.c obmalloc_state_main.  For other interpreters, it is
+    // heap allocated by _PyMem_init_obmalloc() and freed when the
+    // interpreter structure is freed.  In the case of a heap allocated
+    // obmalloc state, it is not safe to hold on to or use memory after
+    // the interpreter is freed. The obmalloc state corresponding to
+    // that allocated memory is gone.  See free_obmalloc_arenas() for
+    // more comments.
+    struct _obmalloc_state *obmalloc;
```

Comment:
There are a couple places where things are simpler if we did not make this a pointer.  That only works, though, if we statically initialize `obmalloc_state_main` and keep the static initializer code in pycore_obmalloc_init.h and pycore_runtime_init.h.

On the other handle we'd be wasting a little bit of memory for the common case (main interpreter).